### ISSUE TITLE
SmartCity-Accessibility Server Side Unit Tests #797 failed

### ISSUE THREAD

**Issue by JuanJenkinsBot** [TechnionYP5777/SmartCity-Accessibility#406]
Build 'SmartCity-Accessibility Server Side Unit Tests' is failing!

Last 50 lines of build :
```
[...truncated 7.22 KB...]
Running smartcity.accessibility.socialnetwork.ReviewCommentTest
Tests run: 5, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.114 sec - in smartcity.accessibility.socialnetwork.ReviewCommentTest
Running smartcity.accessibility.socialnetwork.HelpfulnessTest
Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0 sec - in smartcity.accessibility.socialnetwork.HelpfulnessTest
Running smartcity.accessibility.socialnetwork.UserFunctionalityTest
Tests run: 4, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.011 sec - in smartcity.accessibility.socialnetwork.UserFunctionalityTest
Running smartcity.accessibility.socialnetwork.BestReviewsTest
Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.014 sec - in smartcity.accessibility.socialnetwork.BestReviewsTest
Running smartcity.accessibility.socialnetwork.UserPinTest
Tests run: 3, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.011 sec - in smartcity.accessibility.socialnetwork.UserPinTest
Running smartcity.accessibility.socialnetwork.DeleteReviewsTest
Tests run: 4, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.323 sec - in smartcity.accessibility.socialnetwork.DeleteReviewsTest
Running smartcity.accessibility.socialnetwork.ScoreTest
Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 4.104 sec - in smartcity.accessibility.socialnetwork.ScoreTest
Running smartcity.accessibility.database.UserManagerTest
#
# There is insufficient memory for the Java Runtime Environment to continue.
# Native memory allocation (mmap) failed to map 65536 bytes for committing reserved memory.
OpenJDK 64-Bit Server VM warning: INFO: os::commit_memory(0x00007f132e120000, 65536, 1) failed; error='Cannot allocate memory' (errno=12)
# An error report file with more information is saved as:
# /var/lib/jenkins/sharedspace/SmartCity-Accessibility/server-side/hs_err_pid13907.log
#
# Compiler replay data is saved as:
# /var/lib/jenkins/sharedspace/SmartCity-Accessibility/server-side/replay_pid13907.log

Results :

Tests in error: 
  UserInformationServiceTest.setup:38 » NestedServlet Request processing failed;...

Tests run: 44, Failures: 0, Errors: 1, Skipped: 1

[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 10:32 min
[INFO] Finished at: 2017-06-04T19:08:35+00:00
[INFO] Final Memory: 31M/207M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.19.1:test (default-test) on project SmartCity-Accessibility: Execution default-test of goal org.apache.maven.plugins:maven-surefire-plugin:2.19.1:test failed: The forked VM terminated without properly saying goodbye. VM crash or System.exit called?
[ERROR] Command was /bin/sh -c cd /var/lib/jenkins/sharedspace/SmartCity-Accessibility/server-side && /usr/lib/jvm/java-8-openjdk-amd64/jre/bin/java -javaagent:/var/lib/jenkins/.m2/repository/org/jacoco/org.jacoco.agent/0.7.8/org.jacoco.agent-0.7.8-runtime.jar=destfile=/var/lib/jenkins/sharedspace/SmartCity-Accessibility/server-side/target/jacoco.exec -jar /var/lib/jenkins/sharedspace/SmartCity-Accessibility/server-side/target/surefire/surefirebooter2765029203114939065.jar /var/lib/jenkins/sharedspace/SmartCity-Accessibility/server-side/target/surefire/surefire8203287961606169482tmp /var/lib/jenkins/sharedspace/SmartCity-Accessibility/server-side/target/surefire/surefire_02326886058734102790tmp
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginExecutionException
Build step 'Invoke top-level Maven targets' marked build as failure
```

Changes since last successful build:
- [Ariel Kolikant] 06a173f064d6f90830837a0711710a5d3edb4054 - well that was easy... close #401

[View full output](http://kapalex.ddns.net:8080/job/SmartCity-Accessibility%20Server%20Side%20Unit%20Tests/797/)

**Comment by Kolikant** [comment]
it would appear that the change of giving each individual user his own mapView is a bit too much. up until now, although we thought that they were given one, I found out they were just given a copy of the static one and changed it. it seems now that it is too heavy, raising the question of what to do now

**Comment by JuanJenkinsBot** [comment]
Build was fixed!

**Comment by Kolikant** [comment]
jenkins...... the build was not fixed as no changes to the server side were made. I think this is a resource problem

**Comment by Kolikant** [comment]
![image](https://cloud.githubusercontent.com/assets/23051198/26764790/6a7b0e86-4976-11e7-99aa-0d578bb3d4a4.png)
it was never a problem with the tests, the change made was that now users all each have their own mapView and that just might be too heavy, as mapViews were fundemantily built for a single one for an entire application. 

I added the method of disposing of mapViews, this should be called when users are logged out, rather than having us saving on all of them, also this is the reason why we would wish to log out users after a while.

**Comment by Kolikant** [comment]
keeping this open in case a similar problem ever pops up, as this is unresolved so far

**Comment by JuanJenkinsBot** [comment]
Build was fixed!

**Comment by Kolikant** [comment]
dammnit jenkins! you are unaasigned

**Comment by Kolikant** [comment]
fine! I'll open my own issue! and it's gonna be waaaaaaaaaaaaaaaaay better than your issue ever was jenkins! you will rue this day! ruuuuuuuuuue this day!
